packages/documentation/components/NavBar.vue

```diff
@@ -86,28 +86,22 @@ export default defineComponent({
 				},
 			],
 			sections: menu.map(
-				(
-					section,
-				): {
-					title: string | null
-					links: Array<{
-						icon: Yoco.Icon
-						title: string
-						path: string
-						isActive: boolean
-					}>
-				} => ({
-					links: section.subsections.map((subsection) => ({
-						icon: subsection.icon,
-						title: subsection.title,
-						path: `/${subsection.path}${
-							subsection.pages.length >= 1 ? `/${subsection.pages[0].path}` : ''
-						}`,
-						isActive:
-							subsection.path === ''
-								? route.value.path === '/'
-								: route.value.path.startsWith(`/${subsection.path}`),
-					})),
+				(section): Kotti.Navbar.Section => ({
+					links: section.subsections.map(
+						(subsection): Kotti.Navbar.SectionLink & { to: string } => ({
+							icon: subsection.icon,
+							isActive:
+								subsection.path === ''
+									? route.value.path === '/'
+									: route.value.path.startsWith(`/${subsection.path}`),
+							title: subsection.title,
+							to: `/${subsection.path}${
+								subsection.pages.length >= 1
+									? `/${subsection.pages[0].path}`
+									: ''
+							}`,
+						}),
+					) as unknown as Kotti.Navbar.Section['links'], // TS doesn’t understand the non-empty arrays without the cast due to the .map
```

Comment:
Kind-of an issue with the non-empty arrays.

Reply:
Do you think we should move away from them then? 

I was only hoping we use them as an extra warning that those are required, but if for every usage, we would need a type cast, then maybe not?


The weird thing is, if the arrays are enforced to be non-empty, the map shouldn't be problematic, unless it's used like a conditional array function (e.g. filter) but I guess at compilation time this is not detectable, ad most likely the ts compiler alone won't figure that out. 

In any case, let's discuss potentially not using them then, the disadvantages outweigh the benefits.

Reply:
I think I found a good solution for this. See `chore(add-zod): Replace .nonempty with .refine(...refinementNonEmpty)`